app/models/sns_event/receive_application_event.rb

```diff
@@ -0,0 +1,87 @@
+module SnsEvent
+  class ReceiveApplicationEvent
+    attr_reader :headers, :request_body, :message, :last_http_status
+
+    def initialize(sns_message:, headers: {})
+      raise ArgumentError, 'Headers must be an Enumerable type' unless headers.respond_to?(:key?)
+
+      @headers = headers
+      @request_body = JSON.parse(sns_message) rescue {}
```

Comment:
Looking great! I wonder if better to shift these to their own memoized method, then we can be more explicit about what we're rescuing and logging as @zheileman points out?

Reply:
I was being lazy init

Reply:
@zheileman added some error output ^^^ catches StandardError because nested deserialising wont throw a JSON::ParserError but instead something like 'no implicit conversion of Hash into String'